### PR TITLE
Dump proven claims to file

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
This solves half of #1277 

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Strategies/Verification.hs [hunk 1]
```diff
@@ -95,6 +103,22 @@ didn't manage to verify a claim within the its maximum number of steps.
 If the verification succeeds, it returns ().
 -}
 
+data StuckVerification patt claim
+    = StuckVerification
+        { stuckPattern :: !patt
```
Comment (ttuegel) [inline kore/src/Kore/Strategies/Verification.hs]: Is it possible to actually return the current proof state here, or only the pattern we got stuck on?

File: src/main/k/working/imp/save-proofs/sum-spec.save-proofs-golden [hunk 2]
```diff
@@ -0,0 +1,128 @@
+claim
```
Comment (ttuegel) [inline src/main/k/working/imp/save-proofs/sum-spec.save-proofs-golden]: Should this be a full Kore definition, so we can have multiple claims?

Comment (virgil-serbanuta) [reply src/main/k/working/imp/save-proofs/sum-spec.save-proofs-golden]: I was just doing this in the next PR, but I guess I could move it here.

Comment (virgil-serbanuta) [reply src/main/k/working/imp/save-proofs/sum-spec.save-proofs-golden]: Done.

**Pull request merged**